torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1213,6 +1213,7 @@ void XLATensor::eye_out(XLATensor& out, xla::int64 lines, xla::int64 cols) {
 void XLATensor::fill_(XLATensor& input, const at::Scalar& value) {
   ir::Value constant =
       GetIrValueForScalar(value, input.shape(), input.GetDevice());
+  constant = ir::ops::ExpandAsDynamicShapes(constant, input.GetIrValue());
```

Comment:
I think you shouldn't pass `input.shape()` to `GetIrValueForScalar` in line above here. What happened is that `GetIrValueForScalar`will a static expand to `input.shape()` which is not necessary.

Reply:
addressed